File: 220817-dailycoding/JavaScript/solution.js
// eslint-disable-next-line import/prefer-default-export
export function descending(n) {
  const number = String(n);

  const descendingArray = [];

  for (let i = 0; i < number.length; i += 1) {
    descendingArray[i] = Number(number.charAt(i));
  }

  descendingArray.sort((a, b) => b - a);

  return descendingArray;
}

export function solution(n) {
  let answer = 0;
  let array = [];

  array = descending(n);

  for (let i = 0; i < array.length; i += 1) {
    answer += String(array[i]);
  }

  return Number(answer);
}

File: 220817-dailycoding/Java/app/src/main/java/Solution.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
  public long solution(long n) {
    String answer = "0";

    String number = String.valueOf(n);

    List<String> answers = new ArrayList<>();

    for(int i =0; i<number.length(); i+=1) {
      answers.add(String.valueOf(number.charAt(i)));
    }

    Collections.sort(answers, Collections.reverseOrder());

    for(int i=0; i<answers.size(); i+=1) {
      answer += answers.get(i);
    }

    return Long.parseLong(answer);
  }
}
